### ISSUE TITLE
PLW0177 has false negatives for strings that represent NaN

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#18596]
### Summary

[`nan-comparison` (PLW0177)](https://docs.astral.sh/ruff/rules/nan-comparison/) doesn’t detect `float` calls that produce NaN if their arguments contain positive or negative signs or white space. Compare [`verbose-decimal-constructor` (FURB157)](https://docs.astral.sh/ruff/rules/verbose-decimal-constructor/), which does detect them.

```console
$ cat >plw0177.py <<'# EOF'
float("\n+nAn\xA0") == float("-NaN ")
# EOF

$ ruff --isolated check plw0177.py --select PLW0177 --preview
All checks passed!
```

### Version

ruff 0.11.13 (5faf72a4d 2025-06-05)

**Comment by ntBre** [comment]
Hopefully we can factor out some of the code from https://github.com/astral-sh/ruff/pull/14098 and/or https://github.com/astral-sh/ruff/pull/14596 to help here.
